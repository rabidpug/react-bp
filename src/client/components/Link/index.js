// @flow

import type { Node, } from 'react';
import React from 'react';

type Props = {
  active: boolean,
  children: Node,
  onClick: Function,
};
const Link: Function = ( {
  active, children, onClick,
}: Props ) => {
  if ( active ) return <span>{children}</span>;

  return (
    <a
      href=''
      onClick={ e => {
        e.preventDefault();

        onClick();
      } }>
      {children}
    </a>
  );
};

export default Link;
