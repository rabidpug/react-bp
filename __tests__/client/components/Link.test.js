import Link from 'Components/Link';
import React from 'react';
import { shallow, } from 'enzyme';

describe(
  'Link', () => {
    it(
      'Should render the Link', () => {
        const wrapper = shallow( <Link /> );

        expect( wrapper ).toMatchSnapshot();
      }
    );
  }
);
