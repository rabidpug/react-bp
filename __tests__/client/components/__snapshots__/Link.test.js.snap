// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Should render the Link 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Link />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Icon
          type={undefined}
        />,
        undefined,
      ],
      "onClick": undefined,
      "style": Object {
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "type": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Icon
            type={undefined}
          />,
          undefined,
        ],
        "onClick": undefined,
        "style": Object {
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "type": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
